### ISSUE TITLE
UI doesn't show tracks on albums that are tagged with a different artist than the album

### ISSUE THREAD

**Issue by darioseidl** [volumio/Volumio2-UI#448]
Hello!

I love Volumio, works great on my Rasperry Pi and Pi-DAC+. :D

However, I have the following problem:
When an album contains songs that are credited to a different artist than the album artist, Volumio doesn't show them when selecting the album.
Let's take, for example, the soundtrack "Le Fabuleux Destin d’Amélie Poulain" by Yann Tiersen.
It contains a song by Al Bowlly and another one by Fréhel (all others by Yann Tiersen).
These two songs don't show up when I select the album in Volumio, as you can see here (track 7 and 14 are missing):
![screenshot_2017-12-04_14-16-41](https://user-images.githubusercontent.com/4629505/33554623-cbca2552-d8fd-11e7-8f6b-e409abf12d3f.png)
The same happens when there's a featured artist and the "feat. artist" is written to the song artist tag and not in the song title (which is done, for example, by the MusicBrainz Picard tagger).

This doesn't seem to be a problem for compilations though, I can select a V/A release and see all songs even when they are all by different artists.

I'm using the lastest release.

**Comment by andreas-henning** [comment]
This bug is specific to the Artists and Genres views only. It affects only albums which Volumio doesn't recognize as compilation albums.

@xipmix, @volumio

Volumio has made the concept of a compilation album explicit. The differences in how how Volumio handles compilation vs. non-compilation albums permeates the mpd controller. It looks like fixing this specific bug would be simple. Unfortunately, I just don't understand why Volumio handles compilation albums the way it does, so I'm hesitant to modify anything.

Here is the offending code:

In: 
```js
ControllerMpd.prototype.listAlbumSongs = function (uri,index,previous)
```
If we're browsing from genres, Volumio searches for an album's tracks like this:
```js
if (compilation.indexOf(albumartist)>-1) {
   var findstring = "find album \"" + albumName + "\" genre \"" + genre + "\" ";
}
else {
   var findstring = "find album \"" + albumName + "\" artist \"" + albumartist + "\" genre \"" + genre + "\" ";
}
```
This looks wrong to me in multiple ways:
1. If the album **is** a compilation album, the query will return tracks matching the genre and the album-name, even if the collection contains two albums with identical album-names from different artists. In other areas Volumio considers this possibility and avoids it, but not here.
2. If the album **is not** a compilation album, the query looks only for tracks where the artist = albumartist. This simply makes no sense to me at all. If we're interested in albums, then IMHO the track artist is irrelevant. The only thing that matters is the album-artist.

This is what I think this should look like:

```js
var findstring = "find album \"" + albumName + "\" albumartist \"" + albumartist + "\" genre \"" + genre + "\" ";
```

What exactly would stop working if we made this change? What exactly would stop working if we eliminated ALL the instances where the mpd controller deals with compilation albums explicitly? 

IMHO all the tracks with the same album-name AND album-artist belong to the same album. Period. It's that simple and shouldn't be made any more complicated. It's up to users to ensure taggers/players can determine what tracks belong to the same album by tagging accordingly. All other taggers/players I'm aware work in this way. Why not Volumio as well?

Obviously, I could have all of this completely wrong since I don't know every player/tagger under the sun, nor am I privy to the ideas/requirements that lead to this solution in the first place. Can anyone provide some insight?

**Comment by darioseidl** [comment]
Thanks for looking into this! I haven't looked at the code myself, but your description is spot on.

I usually start with the Artist view, browsing artist then album (this is bugged). Going directly to the Album view works fine.

**Comment by andreas-henning** [comment]
@darioseidl 
Thanks for the feedback! Always good to know. Also that last bit about volumio not adding the full album to the play queue.

I'm wait for feedback on this from somebody "in the know." If I'm not completely off and others think it's a good idea, I'd be glad to eliminate the special handling of compilation albums and attempt to make volumio handle all albums in the same way, based only on album-name and album-artist.

**Comment by nickp1972** [comment]
Can I add a +1 to this issue? Hopefully someone will come along and let Andreas produce this fix. Personally speaking, it's the only blemish on a brilliant music system.

**Comment by robcee** [comment]
is this still open?

**Comment by hiisukun** [comment]
Still an issue for me. I had a dozen or more albums where some of the 'artist' listings for tracks are different to the main albumartist. For example, Bonobo - Days To Come, has several tracks by 'Bonobo feat. Bajka', but most are just by 'Bonobo'.

This breaks: 
Genre search view, once you find an album (it only lists some of the album tracks)
Artist search view (same)
Album search view, once you click the big play button, or use 'Play all' (only plays some album tracks)

I've edited this file: `/app/volumio/plugins/music_service/mpd/index.js` and got everything working as planned, by mostly ignoring the section that check if a compilation is being played or not.

Here's what I came up with for my current version of volumio:
https://gist.github.com/hiisukun/c5386d09e3a6a5e174ed55d9e09941fa

I'll see if I can make a pull request on the relevant repo, without posting the whole file.
